### PR TITLE
Improve Promo Code administration page #863

### PR DESCRIPTION
**Pull request by kkrkuldeep**
Add Attribute into the table
Daniel

### CODE REVIEW

File: src/main/java/alfio/manager/ExtensionManager.java [hunk 1]
```diff
@@ -165,7 +165,8 @@ void handleTicketCancelledForEvent(Event event, Collection<String> ticketUUIDs)
         syncCall(ExtensionEvent.TICKET_CANCELLED, event, payload, Boolean.class);
     }
 
-    void handleOfflineReservationsWillExpire(Event event, List<TicketReservationInfo> reservations) {
+    void
+    handleOfflineReservationsWillExpire(Event event, List<TicketReservationInfo> reservations) {
```
Comment (syjer) [inline src/main/java/alfio/manager/ExtensionManager.java:169]: why the new line?

File: src/main/java/alfio/model/PromoCodeDiscount.java [hunk 2]
```diff
@@ -131,4 +131,56 @@ public static String format(PromoCodeDiscount discount, String eventCurrencyCode
         }
         return MonetaryUtil.formatCents(discount.getDiscountAmount(), eventCurrencyCode);
     }
+
+    public int getId() {
```
Comment (syjer) [inline src/main/java/alfio/model/PromoCodeDiscount.java:135]: no need for all the getters, we are using Lombok for auto generating them. You can remove them.

File: src/main/java/alfio/model/modification/PromoCodeDiscountModification.java [hunk 3]
```diff
@@ -99,4 +99,60 @@ public int getDiscountValue(String currencyCode) {
         }
         return getDiscountInCents(currencyCode);
     }
+
+    public Integer getOrganizationId() {
```
Comment (syjer) [inline src/main/java/alfio/model/modification/PromoCodeDiscountModification.java:103]: No need for the getters, they are auto generated by Lombok. You can remove them.

**Comment by kkrkuldeep** [comment]
Ok I will Remove these things...

File: src/main/java/alfio/util/ImageUtil.java [hunk 4]
```diff
@@ -43,11 +43,13 @@ private ImageUtil() {
 
     public static byte[] createQRCode(String text) {
         try {
+            System.out.println("Inside Try createQRCode");
```
Comment (syjer) [inline src/main/java/alfio/util/ImageUtil.java:46]: remove System.out, don't submit test code

File: src/main/java/alfio/util/ImageUtil.java [hunk 5]
```diff
@@ -43,11 +43,13 @@ private ImageUtil() {
...
             ByteArrayOutputStream baos = new ByteArrayOutputStream();
             BitMatrix matrix = drawQRCode(text);
             MatrixToImageWriter.writeToStream(matrix, "png", baos);
             return baos.toByteArray();
         } catch (WriterException | IOException e) {
+            System.out.println("Inside catch createQRCode");
```
Comment (syjer) [inline src/main/java/alfio/util/ImageUtil.java:52]: remove System.out.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 6]
```diff
@@ -11,30 +11,46 @@ <h1>Create new Event</h1>
             <i class="fa fa-copy"></i> Copy from previous event
         </button>
     </div>
-    <form role="form" name="editEvent" data-ng-submit="save(editEvent, event)" data-error-sensitive="editEventHeader,editPrices" novalidate>
-        <edit-event-header data-target-obj="event" data-organizations="organizations" data-full-edit-mode="true" data-show-existing-errors="showExistingErrors" data-all-languages="allLanguages" data-all-languages-mapping="allLanguagesMapping"></edit-event-header>
+    <form role="form" name="editEvent" data-ng-submit="save(editEvent, event)"
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:14]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 7]
```diff
@@ -11,30 +11,46 @@ <h1>Create new Event</h1>
...
+        <edit-event-header data-target-obj="event" data-organizations="organizations" data-full-edit-mode="true"
+                           data-show-existing-errors="showExistingErrors" data-all-languages="allLanguages"
+                           data-all-languages-mapping="allLanguagesMapping"></edit-event-header>
 
         <div>
             <div class="page-header">
                 <h3>Seats and payment info</h3>
             </div>
-            <edit-prices data-target-obj="event" data-full-edit-mode="true" data-allowed-payment-proxies="allowedPaymentProxies" data-show-existing-errors="showExistingErrors"></edit-prices>
+            <edit-prices data-target-obj="event" data-full-edit-mode="true"
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:24]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 8]
```diff
@@ -11,30 +11,46 @@ <h1>Create new Event</h1>
...
+                         data-allowed-payment-proxies="allowedPaymentProxies"
+                         data-show-existing-errors="showExistingErrors"></edit-prices>
 
             <div class="page-header">
                 <h3>Categories</h3>
-                <small class="text-muted">Here you can define all the ticket categories for the event. Please note that at least one category is required in order to create the event.</small>
+                <small class="text-muted">Here you can define all the ticket categories for the event. Please note that
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:30]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 9]
```diff
@@ -11,30 +11,46 @@ <h1>Create new Event</h1>
...
+                    at least one category is required in order to create the event.</small>
             </div>
 
             <div class="row wMarginBottom" ng-if="event.ticketCategories.length > 0">
                 <div class="col-sm-12">
                     Allocated Seats:
                     <uib-progress max="event.availableSeats || 0">
-                        <uib-bar data-ng-repeat="category in event.ticketCategories | boundedCategories | orderBy: ['inception', 'expiration']" value="calcBarValue(category)" type="{{evaluateBarType($index)}}"><strong>{{category.name}}</strong> ({{category.maxTickets}})</uib-bar>
-                        <uib-bar data-ng-if="(event.ticketCategories | boundedCategories:true).length > 0" value="calcDynamicTickets(event.availableSeats, event.ticketCategories)"><strong>{{allDynamicCategories(event.availableSeats, event.ticketCategories)}}</strong> ({{calcDynamicTickets(event.availableSeats, event.ticketCategories)}})</uib-bar>
+                        <uib-bar
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:38]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 10]
```diff
@@ -11,30 +11,46 @@ <h1>Create new Event</h1>
...
+                        <uib-bar data-ng-if="(event.ticketCategories | boundedCategories:true).length > 0"
+                                 value="calcDynamicTickets(event.availableSeats, event.ticketCategories)"><strong>{{allDynamicCategories(event.availableSeats,
+                            event.ticketCategories)}}</strong> ({{calcDynamicTickets(event.availableSeats,
+                            event.ticketCategories)}})
+                        </uib-bar>
                     </uib-progress>
                 </div>
             </div>
-            <div data-ng-form="category" data-ng-repeat="ticketCategory in event.ticketCategories | orderBy:['ordinal', 'inception.date','inception.time']" ng-class="'category-'+$index">
+            <div data-ng-form="category"
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:51]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 11]
```diff
@@ -50,24 +66,37 @@ <h3>Categories</h3>
...
-                <span ng-if="!canAddCategory(event.ticketCategories)">Please define the number of seats (Max Tickets)</span>
+                <span
+                    ng-if="!canAddCategory(event.ticketCategories)">Please define the number of seats (Max Tickets)</span>
             </div>
 
 
             <div class="row">
                 <div class="col-xs-12">
-                    <button type="button" class="btn btn-success" data-ng-disabled="!canAddCategory(event.ticketCategories)" data-ng-click="addCategory()"><i class="fa fa-plus"></i> Add new</button>
+                    <button type="button" class="btn btn-success"
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:76]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 12]
```diff
@@ -50,24 +66,37 @@ <h3>Categories</h3>
...
-        <div class="row">
-            <div class="col-xs-12 col-md-10 col-md-push-2">
-                <control-buttons data-form-obj="editEvent"></control-buttons>
-            </div>
-            <div class="col-xs-12 col-md-2 col-md-pull-10">
-                <button class="btn btn-lg btn-block btn-default" ng-click="reset()" type="button">Reset values</button>
+            <hr class="wMarginTop30px"/>
+
+            <!--        Adding terms and condition here-->
+            <input type="checkbox" required> I have read and agree to the terms and condition
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:86]: what's that code?

File: src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-event-header.html [hunk 13]
```diff
@@ -1,6 +1,6 @@
 <div data-ng-form="editEventHeader">
     <div class="page-header">
-        <h3>Basic info</h3>
+        <h3>Basic infoz</h3>
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-event-header.html:3]: typo, don't change code if it does not concern the issue you want to fix

File: src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-prices.html [hunk 14]
```diff
@@ -4,12 +4,14 @@
         <div class="form-group">
             <div class="radio">
                 <label>
-                    <input type="radio" name="freeOfCharge" data-ng-model="obj.freeOfCharge" data-ng-value="false"> Entry fee requested
+                    <input type="radio" name="freeOfCharge" data-ng-model="obj.freeOfCharge" data-ng-value="false">
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-prices.html:7]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-prices.html [hunk 15]
```diff
@@ -4,12 +4,14 @@
...
+                    Entry fee requested
                 </label>
             </div>
             <div class="radio">
                 <label>
-                    <input type="radio" name="freeOfCharge" data-ng-model="obj.freeOfCharge" data-ng-value="true"> Free of charge
+                    <input type="radio" name="freeOfCharge" data-ng-model="obj.freeOfCharge" data-ng-value="true"> Free
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-prices.html:13]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/js/admin/feature/promo-codes/edit-promo-code-modal.html [hunk 16]
```diff
@@ -26,7 +25,8 @@ <h3 data-ng-if="promocode.codeType === 'ACCESS'">Insert new Access Code</h3>
 
             <div class="form-group" ng-if="promocode.codeType == 'DISCOUNT'">
                 <label class="control-label" for="discountType">Discount type</label>
-                <select class="form-control" id="discountType" ng-options="key as value for (key , value) in discountTypes" ng-model="promocode.discountType"></select>
+                <select class="form-control" id="discountType"
```
Comment (syjer) [inline src/main/webapp/resources/js/admin/feature/promo-codes/edit-promo-code-modal.html:28]: what has changed here? Don't reformat code please.

File: src/main/webapp/resources/js/admin/feature/promo-codes/promo-codes.html [hunk 17]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (syjer) [inline src/main/webapp/resources/js/admin/feature/promo-codes/promo-codes.html:1]: Don't reformat code please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html (hunk 12 above)
Comment (kkrkuldeep) [reply src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:78]: Working on a different issues.

Comment (syjer) [reply src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html]: If you work on a different issue, please do it in a separate branch, I can't accept a PR that fixes multiple issues.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 18]
```diff
@@ -13,13 +13,11 @@ <h1>Create new Event</h1>
     </div>
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:13]: there are only formatting change here. Revert please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html [hunk 19]
```diff
@@ -50,7 +52,8 @@ <h3>Categories</h3>
             <div class="alert alert-danger" ng-if="event.ticketCategories.length === 0">
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/edit-event.html:52]: there are only formatting change here. Revert please.

File: src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-prices.html [hunk 20]
```diff
@@ -4,7 +4,7 @@
         <div class="form-group">
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/event/fragment/edit-prices.html:4]: there are only formatting change here. Revert please.

File: src/main/webapp/resources/angular-templates/admin/partials/form/control-buttons.html [hunk 21]
```diff
@@ -5,4 +5,4 @@
     <div class="col-md-4 col-md-pull-4 col-xs-12">
```
Comment (syjer) [inline src/main/webapp/resources/angular-templates/admin/partials/form/control-buttons.html:5]: there are only formatting change here. Revert please.

File: src/main/webapp/resources/js/admin/feature/promo-codes/edit-date-promo-code-modal.html [hunk 22]
```diff
@@ -29,6 +29,27 @@ <h3>Edit {{promoCodeDescription}} code settings</h3>
             <input class="form-control" id="description" ng-model="promocode.description" name="description" maxlength="1024">
         </div>
 
+        <!--    Kuldeep Singh-->
```
Comment (syjer) [inline src/main/webapp/resources/js/admin/feature/promo-codes/edit-date-promo-code-modal.html:32]: what are you trying to do here?

File: src/main/webapp/resources/js/admin/feature/promo-codes/edit-promo-code-modal.html [hunk 23]
```diff
@@ -12,7 +12,6 @@ <h3 data-ng-if="promocode.codeType === 'ACCESS'">Insert new Access Code</h3>
                 <span id="helpBlock" class="help-block">Allowed characters: Letters, numbers, and <code>-:_@!$*,;</code></span>
```
Comment (syjer) [inline src/main/webapp/resources/js/admin/feature/promo-codes/edit-promo-code-modal.html:12]: there are only formatting change here. Revert please.

**Pull request closed without merging**